### ISSUE TITLE
upgrades

### ISSUE THREAD

**Issue by gedw99** [chapar-rest/chapar#34]
**Is your feature request related to a problem? Please describe.**

Once the cross platform builds are working, it would be good to add a self update facility.

**Describe the solution you'd like**

The GUI can poll or get a web hook ( from some server somewhere) and then tell the user to upgrade.
Would be good to also have a forced update for hot patches.

**Describe alternatives you've considered**

I dont know if using GitHub as the source of the binaries is best or not.  
I would presume that making it independent of GitHub would be best ? 

**Additional context**

None at this stage. 

There are a few different approach to this, so it's worth a discussion.

**Comment by mirzakhany** [comment]
I agree this is a nice feature to have. 👍

**Comment by gedw99** [comment]
https://github.com/sanbornm/go-selfupdate

quite clever, as it also has binary diff compare.

**Comment by ralyodio** [comment]
I disagree, updates should happen when you use your package manager to update packages.

**Comment by mirzakhany** [comment]
I think notifying the user about a new release its fine but I would prefer to let user decide if they want to update or not.
